docs-client/package.json

```diff
@@ -13,6 +13,7 @@
     "@material-ui/core": "^4.12.4",
     "@material-ui/icons": "^4.11.3",
     "@material-ui/lab": "^4.11.3-deprecations.1",
+    "@monaco-editor/react": "^4.4.6",
     "codemirror": "^5.65.2",
```

Comment:
I think it's worth it if it can replace codemirror enough 👍 
How about removing codemirror ? 🤔

Reply:
I will take a look at this some other time, still working on `JSON Schema` converter when I have time 🙂

Reply:
JSON Schema stuff is taking too much of my time, how about creating a separate issue on deprecating codemirror and handle that in a separate PR?

Reply:
Sorry for the late reply 🙏 

  When I applied the `codemirror`, I had to take into account that `core.jar` might grow too large. Of course, as long as the dependencies are lightweight, it doesn't matter.

Reply:
By any chance, I can compare the size difference in `core.jar`? I was hoping CI to report the size difference 🙂

Reply:
In addition, there is a bug related to it.
https://github.com/line/armeria/issues/4500

Reply:
I remember checking using [webpack-bundle-analyzer](https://www.npmjs.com/package/webpack-bundle-analyzer). 
- https://github.com/line/armeria/issues/3706

Reply:
I completely miss this context. 😅 
@heowc So we don't need the codemirror? or should we replace it with another one?

Reply:
Monaco editor should be able to replace code-mirror but it just needs some testing done which I can't find time to do yet 😆

Reply:
I got it. We don't have to replace it right now. 😄

Reply:
Sorry for the very late response. 😭 

I was wondering if it would be possible to change the graphql editor as well.
As a side note, when we added code-mirror as a graphql editor, we were also thinking about monaco, but we weren't mature enough to do it with monaco yet and chose code-mirror.

The reason I mention this is that I remember that the editor itself took up a surprisingly large amount of space. If this is acceptable, I think it's worth it.

Reply:
Whichever you folks choose, I hope we choose something more future-proof, although things change really quick in front-end world. 😆

Reply:
As I am on-call for this week, I couldn't find much time to look into this issue. I was primarily AFK in my free time 😅 

The intention of this PR was to introduce autocomplete & schema validation, if the CodeMirror editor can do that, not introducing a change on the editor might be better in the short run. I am not a front-end expert either, so I don't have strong opinion on either editors 🥶  

I will test and try to come up with a solution of weekend.

Reply:
Sorry about the late response,

I have experimented with the CodeMirror and I couldn't find a way to set JSON schemas with the current version of CodeMirror we are using (which is https://github.com/scniro/react-codemirror2 and last published 3 years ago 😢 ).


So, I can continue working on migration to the `Monaco` editor if you don't have any objections. If we want to keep using CodeMirror I suppose we should upgrade to `CodeMirror 6` which contains a lot of breaking changes and we should remove the `react-codemirror2` dependency.

So, the question is `CodeMirror 6` upgrade vs `Monaco` editor? I will try upgrading CodeMirror to 6. Based on this read: https://blog.replit.com/code-editors I feel like we can just upgrade to code mirror 6. Also `GraphiQL` is written in code mirror so our change for the GraphQL editor shouldn't be huge.

Reply:
2nd update,

I have played with `Codemirror 6` a little bit. Unfortunately, there were some issues:
- Webpack did not work properly out of the box, I had to change the config of webpack to add some aliases.
- JSON language with Schemas is not supported by default and trying to implement a custom language plugin though.
- No graphQL support out of the box as well but need to install external dependencies which makes bundle size grow.

I think the biggest take-away is point number 2. We can't just have JSON schema support easily. Therefore I am now playing with Monaco which is bundled with both JSON Schema support and GraphQL support out of the box. Considering the team doesn't focus on front-end related features (makes sense as this is just a helper page for devs) I think we should choose Monaco over CodeMirror 6 for ease-of use and the features it provides out-of-the box.

Reply:
Have you considered `monaco-graphql`?
- https://github.com/graphql/graphiql/tree/main/packages/monaco-graphql

Reply:
Yes, I tried using it. But I couldn't use it with the 2 most popular monaco/react libraries. (Monaco-editor/react and react-monaco-editor) Probably there is a way to use it I need to spend some more time to play with webpack config.

Reply:
I pushed my latest changes, and would be more than happy to pair with someone on this topic.

Reply:
I have tried disabling some of the features form the web pack Monaco plugin but couldn't succeed. Also, it seems like different workers share some common files but for some reason, they seem to be duplicated.

Reply:
Seems like the size is the same, 2 MBs should be fine right?

Reply:
Good question. I think it is an inevitable choice at the moment. Many users would prefer rich features although the packaging size has slightly increased. After using autocompletion, I didn't want to get back. 😉 We may consider splitting `DocService` from the core module and providing it as an opt-in feature in the next major version.

Reply:
Opened an issue so this won't be forgotten
- https://github.com/line/armeria/issues/4795#issue-1648378398

Closing the conversation 😉